regtech_mail_api/api.py

```diff
@@ -0,0 +1,66 @@
+from fastapi import FastAPI, Request
+from regtech_mail_api.mailer import Mailer, MockMailer, SmtpMailer
+
+from regtech_mail_api.models import Email
+from regtech_mail_api.settings import EmailApiSettings, EmailMailerType
+
+app = FastAPI()
+
+# FIXME: Come up with a better way of handling these settings
+#        without having to do all the `type: ignore`s
```

Comment:
I was able to remove the # type: ignores and run fine in docker-compose.  If the issue is with potentially running the service without env vars used, what about setting smtp_host, username and password to str = "local" or something similar instead of None?  For instance, I was able to update my local copy of the settings to:
    email_mailer: EmailMailerType = EmailMailerType.SMTP
    smtp_host: str = "local"
    smtp_port: int = 0
    smtp_username: SecretStr = "local"
    smtp_password: SecretStr = "local"
    smtp_use_tls: bool = False
    from_addr: EmailStr = "test@test.com"
    to: EmailStr = "test@test.com"
    cc: set[EmailStr] | None = None
    bcc: set[EmailStr] | None = None
    
    and then run `docker run -it -d --name mailing-api3 -p 8765:8765 -e email_mailer='mock' mail-image` and ran the curl command in the readme and it worked fine.

Reply:
Yeah, it all runs fine without the `type: ignore`s in there, it's just the python type checker that's grumpy. 

<img width="1381" alt="image" src="https://github.com/cfpb/regtech-mail-api/assets/4269085/aba9ddfd-2649-4bc7-923e-7efe51a69c97">

Giving them all default values does fix the type checker, but then you lose some of the validation magic [pydantic-settings](https://docs.pydantic.dev/latest/concepts/pydantic_settings/) gives you around for checking required unset values. We _could_ still go the default-all-the-things route and just add more validator functions. 🤷

Reply:
Gotcha, I’ll have to do some Googling for niftier ways of resolving that